PcdcAnalysisTools/blueprint/routes/views/tableOne/__init__.py

```diff
@@ -0,0 +1,261 @@
+            base_field = nested_fields[0]  # The column name (e.g., 'survival_characteristics')
+            target_field = nested_fields[-1]  # The field to extract (e.g., 'lkss_obfuscated')
+            # Check if the column exists
+            if base_field in df.columns:
+                # Handle lists of dictionaries - extract the first value that matches
+                df[field] = df[base_field].apply(
+                    lambda x: x[0].get(target_field) if isinstance(x, list) and len(x) > 0 and isinstance(x[0], dict) else None
+                )
+                
+                # Drop the original column if requested
+                df = df.drop(base_field, axis='columns')
+
+    return df
+
+
+def _check_user_input(args):
+    
+    config = capp.config.get("TABLE_ONE", DEFAULT_TABLE_ONE_CONFIG)
+
+    filter_sets = args.get("filterSets")
+    covariates = args.get("covariates")
+    #for first version only allow one filter set
+    if not filter_sets:
+        raise UserError("You must submit a filter set")
+    if len(filter_sets) != 1:
+        raise UserError("You must submit exactly one filter set")
+
+    if not covariates:
+        raise UserError("You must submit at least one covariate")
+    
+    for name, covariate in covariates.items():
+        if covariate["type"] not in ["continuous", "categorical"]:
+            raise UserError(
+                f"Unsupported covariate type: {covariate['type']}. Supported types are 'continuous' and 'categorical'."
+            )
+        if covariate["type"] == "categorical":
+            if "selectedKeys" not in covariate or not covariate["selectedKeys"]:
+                raise UserError(
+                    f"You must provide selectedKeys for categorical covariate: {name}"
+                )
+        if covariate["type"] == "continuous":
+            #add this in later
+            # if "unit" not in covariate or not isinstance(covariate["unit"], (int, float)):
+            #     raise UserError(
+            #         f"You must provide a numeric unit for continuous covariate: {name}"
+            #     )
+            if "selectedKeys" in covariate:
+                raise UserError(
+                    f"You cannot provide selectedKeys for continuous covariate: {name}"
+                )
+
+    # TODO - we want to add custom logs for tracking usage and abse like in the survival
+
+    log_obj = {}
+    if not capp.mock_data:
+        try:
+            user = auth.get_current_user()
+            log_obj["user_id"] = user.id
+        except AuthError:
+            logger.warning(
+                "Unable to load or find the user, check your token"
+            )
+    for filter_set in filter_sets:
+        try:
+            log_obj["filter"] = filter_set["filter"]
+            # log_obj["explorer_id"] = filter_set["explorerId"]
+            log_obj["filter_name"] = filter_set["name"]
+            log_obj["filter_set_id"] = filter_set["id"]
+            logger.info(log_obj)
```

Comment:
could you add a "TABLE ONE -" string at the beginning of the log for searcheability like in the survival curve: https://github.com/chicagopcdc/PcdcAnalysisTools/blob/f1c1c1fcc68f1bb75698daa21278ebbb36dd2017/PcdcAnalysisTools/blueprint/routes/views/survival/__init__.py#L50C5-L50C63